cobra.go

```diff
@@ -39,6 +46,20 @@ var MousetrapHelpText string = `This is a command line tool
 You need to open cmd.exe and run it from there.
 `
 
+//AddTemplateFunc adds a template function that's available to Usage and Help
+//template generation.
+func AddTemplateFunc(name string, tmplFunc interface{}) {
```

Comment:
I was going to say, it's not *my* choice. It's golang's choice to persist a void pointer instead of implementing generics!